main.go

```diff
@@ -42,81 +46,106 @@ func main() {
 		os.Exit(2)
 	}
 
-	buf := bufio.NewReader(os.Stdin)
-	var regArr []*regexp.Regexp
-
-	if *fixedStringsParameter {
-		for i := range flagTest {
-			flagTest[i] = regexp.QuoteMeta(flagTest[i])
-		}
+	if len(flag.Args()) == 0 {
+		loopCount = 1
```

Comment:
A loop even with a hardcoded number of rounds? C'mon. Make a function taking a `*File`.